mailpoet/assets/js/src/mailpoet.ts

```diff
@@ -1,8 +1,17 @@
-import FeaturesController from 'features_controller';
-import MailPoetComUrlFactory from 'mailpoet_com_url_factory';
-import { MailPoetDate as Date } from 'date';
+import FeaturesController from './features_controller';
+import MailPoetComUrlFactory from './mailpoet_com_url_factory';
+import { MailPoetI18n } from './i18n';
+import { MailPoetDate } from './date';
+import { MailPoetAjax } from './ajax';
+import { MailPoetModal } from './modal';
+import { MailPoetNotice } from './notice';
+import { MailPoetForceTrackEvent, MailPoetTrackEvent } from './analytics_event';
+import { MailPoetNum } from './num';
+import { MailPoetHelpTooltip } from './help-tooltip';
+import { MailPoetIframe } from './iframe';
+
 // A placeholder for MailPoet object
-var MailPoet = {
+export const MailPoet = {
```

Comment:
It's usually super hard to convince me to use type casting, but unfortunately `as object` serves a bit more than type-casting, but I treat casting as a bad thing and deceitful to TS, anything that would prevent TS's type-inference feels malicious.

Reply:
@samnajian This is not type-casting (although I agree it may look like one). The `as const` is a [special assertion](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) that's super-useful for objects like these. It infers the exact property values (not just general types) and makes the properties readonly. Indeed, it just results in more exact type definition of what the object really is.

Also see this playground: https://www.typescriptlang.org/play?#code/MYewdgzgLgBCBWAjAjDAvDA3jA5AQxwC5ccAaGRYzAXxmoG4AoUSWEReAJnS1wOJxkKVWrTwQYLaE0YIUsjp0ZA

And the difference between `obj1` and `obj2` in `d.ts`:
```ts
declare const ojb1: {
    a: string;
    b: {};
};
declare const obj2: {
    readonly a: "";
    readonly b: {};
};

```

You can see that in `obj1` the `a` property is inferred to be a string, while in `obj2` it's (more correctly), the exact `""` string.

Reply:
Yea checked and got the reason for suggesting this change that's why I'm saying `serves a bit more than type-casting` which means I'll do it, but it looks verrryyyyyy much like a casting and can result in developers think it's casting and promote wrong culture, I'm thinking about a way to make it explicit that this is not a type-casting, but hard to fine one  😅

Reply:
> but it looks verrryyyyyy much like a casting and can result in developers thinking it's casting and promote wrong culture, I'm thinking about a way to make it explicit that this is not a type-casting, but hard to find one

I think no need to make it explicit or anything, it's a very common TS construct.